nexus/db-queries/src/db/saga_recovery.rs

```diff
@@ -285,17 +303,38 @@ async fn load_saga_log(
     // Although we could read them all into memory simultaneously, this
     // risks blocking the DB for an unreasonable amount of time. Instead,
     // we paginate to avoid cutting off availability.
-    let mut last_id = None;
     let mut events = vec![];
-    loop {
-        let pagparams = new_page_params(last_id.as_ref());
-        let mut some_events =
-            datastore.saga_node_event_list_by_id(saga.id, &pagparams).await?;
-        if some_events.is_empty() {
-            break;
-        }
-        events.append(&mut some_events);
-        last_id = Some(events.last().as_ref().unwrap().saga_id.0);
+    let mut paginator = Paginator::new(BATCH_SIZE);
+    let conn = datastore.pool_connection_unauthorized().await?;
+    while let Some(p) = paginator.next() {
+        use db::schema::saga_node_event::dsl;
+        let batch = paginated_multicolumn(
+            dsl::saga_node_event,
+            (dsl::node_id, dsl::event_type),
+            &p.current_pagparams(),
+        )
+        .filter(dsl::saga_id.eq(saga.id))
+        .select(db::saga_types::SagaNodeEvent::as_select())
+        .load_async(&*conn)
+        .map_err(|e| {
+            public_error_from_diesel(
+                e,
+                ErrorHandler::NotFoundByLookup(
+                    ResourceType::SagaDbg,
+                    LookupType::ById(saga.id.0 .0),
+                ),
```

Comment:
I think this is what the old code was doing but I don't think it's right?  It may not matter.  But if we get no rows here, that's not an error at all.

Reply:
I added a test for this case, I think the "no node, no saga in db" case still returns a vec of length zero, so I'm pretty sure this is dead code. Updated to `ErrorHandler::Server` to make this slightly more apparent.

Reply:
Just to be super sure, I did a quick check and it looks to me like the `ErrorHandler::NotFound*` variants only do anything special if the returned Diesel error variant is itself [`NotFound`](https://docs.rs/diesel/latest/diesel/result/enum.Error.html#variant.NotFound), which only happens if we use `first()` (`first_async()` for us) or `get_result()` (`get_result_async()` for us).  We're using `load_async()`, and the `NotFound` docs explicitly say `load()` doesn't consider it an error to get 0 rows.  So that explains why it didn't matter and yes I think the update you made makes sense.  Thanks!